src/xml_interface.F90

```diff
@@ -40,492 +33,300 @@ module xml_interface
 contains
 
 !===============================================================================
-! OPEN_XMLDOC opens and parses an XML and returns a pointer to the document
-!===============================================================================
-
-  subroutine open_xmldoc(ptr, filename)
-
-    character(len=*) :: filename
-    type(Node), pointer :: ptr
-
-    ptr => parseFile(trim(filename))  ! Pointer to the whole XML document
-    ptr => getDocumentElement(ptr)    ! Grabs root element of XML document
-
-  end subroutine open_xmldoc
-
-!===============================================================================
-! CLOSE_XMLDOC closes and destroys all memory associated with document
-!===============================================================================
-
-  subroutine close_xmldoc(ptr)
-
-    type(Node), pointer :: ptr
-
-    ptr => getParentNode(ptr)  ! Go from root element ptr to document ptr
-    call destroy(ptr)          ! Deallocates all nodes recursively
-
-  end subroutine close_xmldoc
-
-!===============================================================================
 ! CHECK_FOR_NODE checks for an attribute or sub-element node with the given
 ! node name. This should only be used for checking a single occurance of a
-! sub-element node. To check for sub-element nodes that repeat, use
-! get_node_list and get_list_size. This is to minimize number of calls
-! to getChildrenByTagName.
+! sub-element node.
 !===============================================================================
 
-  function check_for_node(ptr, node_name) result(found)
-
-    character(len=*), intent(in) :: node_name
+  function check_for_node(node, name) result(found)
+    type(XMLNode), intent(in) :: node
+    character(len=*), intent(in) :: name
     logical :: found
-    type(Node), pointer, intent(in) :: ptr
-
-    type(Node), pointer :: temp_ptr
-    type(NodeList), pointer :: elem_list
-
-    ! Default that we found it and attribute
-    found = .true.
 
-    ! Get the attribute node
-    temp_ptr => getAttributeNode(ptr, trim(node_name))
+    type(XMLAttribute) :: attr
+    type(XMLNode) :: child
 
-    ! Check if node exists
-    if (associated(temp_ptr)) return
-
-    ! Check for a sub-element
-    elem_list => getChildrenByTagName(ptr, trim(node_name))
-
-    ! Get the length of the list
-    if (getLength(elem_list) == 0) then
-      found = .false.
-      return
+    ! Check if an attribute or  exists
+    attr = node%attribute(name)
+    if (attr%associated()) then
```

Comment:
I was handily outvoted and our style guide [now mandates](http://openmc.readthedocs.io/en/latest/devguide/styleguide.html#whitespace-in-expressions) using spaces around `%`. Thanks for keeping me honest; I'll fix this.